### PR TITLE
txm: generalizing resender

### PR DESCRIPTION
**Pull request by aalu1418**
ticket: https://smartcontract-it.atlassian.net/browse/BCI-862

### CODE REVIEW

File: core/chains/evm/txmgr/eth_resender.go
Comment (prashantkumar1982) [inline core/chains/evm/txmgr/eth_resender.go:1]: nit: rename the file too to resender.go?

File: core/chains/evm/txmgr/eth_resender.go [hunk 1]
```diff
@@ -60,20 +58,41 @@ type EthResender[
 func NewEthResender(
```
Comment (prashantkumar1982) [inline core/chains/evm/txmgr/eth_resender.go:58]: nit: rename to NewEvmSender

File: core/chains/evm/txmgr/eth_resender.go [hunk 2]
```diff
@@ -60,20 +58,41 @@ type EthResender[
...
 	lggr logger.Logger,
 	txStore EvmTxStore,
-	ethClient evmclient.Client, ks EvmKeyStore,
+	ethClient EvmTxmClient,
```
Comment (prashantkumar1982) [inline core/chains/evm/txmgr/eth_resender.go:61]: nit: rename to evmClient

File: core/chains/evm/txmgr/txmgr.go [hunk 3]
```diff
@@ -95,7 +95,7 @@ type Txm[
 	wg       sync.WaitGroup
 
 	reaper           *Reaper[CHAIN_ID]
-	ethResender      *EthResender[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE, R, ADD]
+	ethResender      *Resender[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE, R, ADD]
```
Comment (prashantkumar1982) [inline core/chains/evm/txmgr/txmgr.go:98]: nit: rename to resender, now that it is generalized.

**Review by prashantkumar1982 — COMMENTED** [review]
Looks good, just some nit renames to reflect what we generalized here.

File: common/txmgr/types/client.go [hunk 4]
```diff
@@ -92,7 +92,7 @@ type TxmClient[
 	) (txReceipt []R, txErr []error, err error)
 	SendEmptyTransaction(
 		ctx context.Context,
-		txAttemptBuilder TxAttemptBuilder[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE, ADD],
+		emptyTxBuilder func(seq SEQ, feeLimit uint32, fee FEE, fromAddress ADDR) (attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], err error),
```
Comment (jmank88) [inline common/txmgr/types/client.go:95]: Is `Builder` overstating now that it's just a func? Is it just a regular constructor?
```suggestion
		newTx func(seq SEQ, feeLimit uint32, fee FEE, fromAddress ADDR) (attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, R, FEE, ADD], err error),
```

**Review by jmank88 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**